### ISSUE TITLE
Column Selection (and Sorting)

### ISSUE THREAD

**Issue by mikecarnohan** [elibroftw/music-caster#71]
Awesome player. Love the low latency... great job! Music Caster would be even better with the ability to display columns from the file (e.g. BPM, Track Length, Bitrate, Album, Track Number) and have each be sortable. With this addition, it would allow a person like myself to replace my go to player (Media Monkey, which I have issues using for streaming). I think to simplify, you could read the file's metadata and display the following default columns (Filename, Artists, Track Name, Album, Track Number, BPM, Bitrate, Length), and allow all to be turned on/off, sorted, and ordered as the user wants. The player may need a slightly wider footprint to accommodate the new information, but  I believe the upgrade would be worth it.

**Comment by elibroftw** [comment]
Like a better library tab?

**Comment by mikecarnohan** [comment]
Yes, but directly in the Queue tab, so that tracks can be played (and sorted) directly. It's not currently clear what purpose the Library tab serves, since it seems unrelated to the Queue, where most of the apps focus rightly is. 

I'd love to connect offline with you to go over some interface ideas if you're open to it (mostly geared at cleaning up the UI, and making it simpler to use, and ideally maintain dev-wise). I also have some small bugs to report (mostly unrelated to the app development, owing to things like antivirus clients blocking the app initially, that could be address in documentation... that I'm also open to helping you with).

You have a great tool... I really hope you'll stick with its development, and make it into a great player (in addition to being a great casting tool). If I can be of any help, please let me know... I'm a UX designer and oversaw operations at a software company for the better part of a decade, and am happy to pitch in. 

The changes needed to optimize the interface are fairly straightforward (move the play controls to the bottom of the player, spanning the entire width, shrink the album art and details column on the left, expand the queue area, move URL under the queue on one row (above the player controls), ditch the library and timer tabs (moving timer options into settings area), allow Meta data to be directly edited in the queue, so the Metadata tab can also go, and changing Settings to a gear icon, and not as a separate tab. These changes would make it more inline with great players (like MediaMonkey) and bring it up to a level that people could use as their primary player. I'm not just trying to create more work for you, or shoot down what you've created... I really am a fan of your work, and would love to see it see wide adoption, since there's really not much out there like it. Thanks for all that you've already done!

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER](https://github.com/elibroftw/music-caster/assets/39128541/c914d8c5-bd82-420b-826f-52c98b145e24)

This is a quick mock-up of how to transform the interface to simply it, and add the newly requested column sorting functionality.

**Comment by mikecarnohan** [comment]
You can update the add to queue dialog to include URL as as option (there are open dialog options that allow for a local file, or a URL), and in that way, rather than having a separate column for URL, you can allow a URL to be added to the general queue (just list the path under a column, like title, and perhaps prepend an icon designating it as URL sourced).

![URL](https://github.com/elibroftw/music-caster/assets/39128541/0dc05ac2-2560-4813-99d7-7d89250b674a)

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER2](https://github.com/elibroftw/music-caster/assets/39128541/be0ef861-047c-40a1-9336-90e78dce61a9)

The alternative would be to put the URL info under the queue, and follow similar logic (to above), and stick URL-added tracks in the queue if they select Add to Queue.

**Comment by mikecarnohan** [comment]
The main goal with these changes is to simply the interface, and focus on functionality that will primarily be used (relegating lesser used functionality, to other areas or other work flows... like allowing save of playlists, but saving them locally, and allowing them to later be added to the player by the general add option, rather than giving them their own tabs, and taking the focus off of it being a media player primarily).

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER3](https://github.com/elibroftw/music-caster/assets/39128541/c59020cc-56ed-47c1-8d5f-fa33726ba6dc)

Here is the mock up in your existing color scheme, with some minor additional tweaks.

One last note... the ID3 edit functionality would be triggered by clicking an icon or edit box to the right of the track details. The edit area below would only appear if the user clicked to edit. This saves real estate, and I think it a bit more intuitive. The only draw back, is that it means they can only edit Now Playing tracks, and possibly presents a technical problem, though ID3 is usually editable while a track is paying in most players. If the above issues are real ones, it can easily be accommodated in subsequent redesign efforts.

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER3](https://github.com/elibroftw/music-caster/assets/39128541/b55a532e-7e4f-4373-9e59-77486d7caa79)

Here is the mock up in your existing color scheme, with some minor additional tweaks.

One last note... the ID3 edit functionality would be triggered by clicking an icon or edit box to the right of the track details. The edit area below would only appear if the user clicked to edit. This saves real estate, and I think it a bit more intuitive. The only draw back, is that it means they can only edit Now Playing tracks, and possibly presents a technical problem, though ID3 is usually editable while a track is paying in most players. If the above issues are real ones, it can easily be accommodated in subsequent redesign efforts.

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER3](https://github.com/elibroftw/music-caster/assets/39128541/0bb2dc6d-502a-4c43-b7be-de02cbdb1776)

Here is the mock up in your existing color scheme, with some minor additional tweaks.

One last note... the ID3 edit functionality would be triggered by clicking an icon or edit box to the right of the track details. The edit area below would only appear if the user clicked to edit. This saves real estate, and I think it a bit more intuitive. The only draw back, is that it means they can only edit Now Playing tracks, and possibly presents a technical problem, though ID3 is usually editable while a track is paying in most players. If the above issues are real ones, it can easily be accommodated in subsequent redesign efforts.

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER3](https://github.com/elibroftw/music-caster/assets/39128541/76ca88a8-2b1c-43b7-8297-460a2ec56b17)

Here is the mock up in your existing color scheme, with some minor additional tweaks.

One last note... the ID3 edit functionality would be triggered by clicking an icon or edit box to the right of the track details. The edit area below would only appear if the user clicked to edit. This saves real estate, and I think it a bit more intuitive. The drawback is that it means they can only edit Now Playing tracks, and it possibly presents a technical problem (though ID3 is editable while a track is paying in many players). If the above issues are real ones, it can easily be accommodated in subsequent redesign efforts.

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER3](https://github.com/elibroftw/music-caster/assets/39128541/3792bce0-1609-4c54-bd8d-6324f446be2c)

Here is the mock up in your existing color scheme, with some minor additional tweaks.

One last note... the ID3 edit functionality would be triggered by clicking an icon or edit box to the right of the track details. The edit area below would only appear if the user clicked to edit. This saves real estate, and I think it a bit more intuitive. The drawback is that it means they can only edit Now Playing tracks, and it possibly presents a technical problem (though ID3 is editable while a track is paying in many players). If the above issues are real ones, it can easily be accommodated in subsequent redesign efforts.

**Comment by mikecarnohan** [comment]
![MUSIC_CASTER5](https://github.com/elibroftw/music-caster/assets/39128541/d7a8a79c-da2a-4934-8235-ddc27b1acd46)

This is one way to handle the timer, if sticking it in Settings or eliminating it both don't appeal.

**Comment by Ecnassianer** [comment]
I would love this version of the library. I have a bunch of different songs with similar names, and being able to see the album etc in each column would be really helpful.
